hs_core/hydroshare/utils.py

```diff
@@ -762,13 +762,15 @@ def get_remaining_user_quota(user_or_username, units='MB'):
 
     if user and user.is_active:
         uq = user.quotas.filter(zone='hydroshare').first()
-        if uq:
-            qmsg = QuotaMessage.objects.first()
-            enforce_flag = qmsg.enforce_quota
-            if enforce_flag:
-                remaining = uq.allocated_value - uq.used_value
-                remaining = convert_file_size_to_unit(remaining, to_unit=units, from_unit=uq.unit)
-                return max(remaining, 0)
+        if not uq:
+            # create a quota object for the user
+            uq = user.quotas.create(zone='hydroshare')
+        qmsg = QuotaMessage.objects.first()
+        enforce_flag = qmsg.enforce_quota
+        if enforce_flag:
+            remaining = uq.allocated_value - uq.used_value
```

Comment:
This looks like the reason why Martin was able to upload a 25 GB file. The enforcement is based on the remaining quota with no hard limit?